contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -463,16 +463,25 @@ contract SmartAccount is
         SafeERC20.safeTransfer(tokenContract, dest, amount);
     }
 
-    function execute(address dest, uint value, bytes calldata func) external onlyOwner{
+    function executeCall(
+        address dest,
+        uint256 value,
+        bytes calldata func
+    ) external nonReentrant {
         _requireFromEntryPointOrOwner();
         _call(dest, value, func);
     }
 
-    function executeBatch(address[] calldata dest, bytes[] calldata func) external onlyOwner{
+    function executeBatchCall(
+        address[] calldata dest,
+        uint256[] calldata value,
+        bytes[] calldata func
+    ) external nonReentrant {
         _requireFromEntryPointOrOwner();
-        require(dest.length == func.length, "wrong array lengths");
-        for (uint i = 0; i < dest.length;) {
-            _call(dest[i], 0, func[i]);
+        require(dest.length == value.length, "wrong array lengths");
```

Comment:
should we explicitly revert when lengths are 0?

Reply:
resolved

Reply:
have removed and added a test